assets/js/phoenix_live_view.js

```diff
@@ -1055,10 +1064,12 @@ class DOMPatch {
 
         if(fromEl.isSameNode(focused) && DOM.isFormInput(fromEl)){
           this.trackBefore("updated", fromEl, toEl)
-          DOM.mergeInputs(fromEl, toEl)
+          DOM.mergeFocusedInput(fromEl, toEl)
+          DOM.syncAttrsToProps(fromEl)
```

Comment:
Another way to think about this is `syncPropToAttr`.  If `fromNode.checked` (but no attribute), then force it on the `toNode.checked`.  In morphdom, we will convert this to an attr later on in the morphing process.

I think this might get around the need for `<%= @checked && "checked" %>` and avoid updating the `.checked` property if we don't need to on the `fromEl`.

https://github.com/patrick-steele-idem/morphdom/pull/177

Reply:
@chrismccord 

https://github.com/patrick-steele-idem/morphdom/pull/177/commits/36360a206a1f940b62e56fc29abc543b287e4c7d

I think this commit outlines what exactly will happen when the `toNode` as a string does not contain any information that it should be checked.  It forces the `fromNode` to turn off checked.